src/main/java/org/opensearch/plugin/insights/core/service/categorizer/QueryShapeGenerator.java

```diff
@@ -264,78 +221,13 @@ private static <T> Map.Entry<Class<?>, List<Function<Object, String>>> createEnt
-            return Map.ofEntries(
-                createEntry(IpRangeAggregationBuilder.class, IpRangeAggregationBuilder::field),
-                createEntry(AutoDateHistogramAggregationBuilder.class, AutoDateHistogramAggregationBuilder::field),
-                createEntry(DateHistogramAggregationBuilder.class, DateHistogramAggregationBuilder::field),
-                createEntry(HistogramAggregationBuilder.class, HistogramAggregationBuilder::field),
-                createEntry(VariableWidthHistogramAggregationBuilder.class, VariableWidthHistogramAggregationBuilder::field),
-                createEntry(MissingAggregationBuilder.class, MissingAggregationBuilder::field),
-                createEntry(AbstractRangeBuilder.class, AbstractRangeBuilder::field),
-                createEntry(GeoDistanceAggregationBuilder.class, GeoDistanceAggregationBuilder::field),
-                createEntry(DiversifiedAggregationBuilder.class, DiversifiedAggregationBuilder::field),
-                createEntry(RareTermsAggregationBuilder.class, RareTermsAggregationBuilder::field),
-                createEntry(SignificantTermsAggregationBuilder.class, SignificantTermsAggregationBuilder::field),
-                createEntry(TermsAggregationBuilder.class, TermsAggregationBuilder::field),
-                createEntry(AvgAggregationBuilder.class, AvgAggregationBuilder::field),
-                createEntry(CardinalityAggregationBuilder.class, CardinalityAggregationBuilder::field),
-                createEntry(ExtendedStatsAggregationBuilder.class, ExtendedStatsAggregationBuilder::field),
-                createEntry(GeoCentroidAggregationBuilder.class, GeoCentroidAggregationBuilder::field),
-                createEntry(MaxAggregationBuilder.class, MaxAggregationBuilder::field),
-                createEntry(MinAggregationBuilder.class, MinAggregationBuilder::field),
-                createEntry(StatsAggregationBuilder.class, StatsAggregationBuilder::field),
-                createEntry(SumAggregationBuilder.class, SumAggregationBuilder::field),
-                createEntry(ValueCountAggregationBuilder.class, ValueCountAggregationBuilder::field),
-                createEntry(ValuesSourceAggregationBuilder.class, ValuesSourceAggregationBuilder::field)
-            );
-        }
-
-        /**
-         * Returns a map where the keys are sort builders, and the values are lists of
-         * functions that extract field values from instances of these classes.
-         *
-         * @return a map with class types as keys and lists of field extraction functions as values.
-         */
-        private static Map<Class<?>, List<Function<Object, String>>> getSortFieldDataMap() {
-            return Map.ofEntries(createEntry(FieldSortBuilder.class, FieldSortBuilder::getFieldName));
+        private static Map<Class<?>, List<Function<Object, String>>> getFieldDataMethodsMap() {
+            return Map.ofEntries(createEntry(WithFieldName.class, WithFieldName::fieldName));
```

Comment:
given we have single instance today, no need to have this unnecessary map. We can include it if needed later